lib/topological_inventory/ansible_tower/logging.rb

```diff
@@ -15,6 +15,15 @@ module Logging
       def logger
         TopologicalInventory::AnsibleTower.logger
       end
+
+      def log_with(request_id)
+        old_request_id = Thread.current[:request_id]
+        Thread.current[:request_id] = request_id
+
+        yield
+
+        Thread.current[:request_id] = old_request_id
```

Comment:
Should this be in an `ensure` block to guarantee that the original request_id is restored?

Reply:
👍 yes, ensure block added